p2p/src/sync/tests/request_response.rs

```diff
@@ -90,34 +90,36 @@ async fn test_multiple_requests_and_responses() {
     assert_eq!(request_ids.len(), 2);
 
     for _ in 0..2 {
-        if let Ok(net::types::SyncingEvent::Request {
-            peer_id: _,
-            request_id,
-            request: _,
-        }) = mgr2.handle.poll_next().await
-        {
-            mgr2.handle
-                .send_response(
-                    request_id,
-                    Response::HeaderResponse(HeaderResponse::new(vec![])),
-                )
-                .await
-                .unwrap();
-        } else {
-            panic!("invalid data received");
+        tokio::select! {
+            event = mgr2.handle.poll_next() => match event {
+                Ok(net::types::SyncingEvent::Request { request_id, .. }) => {
+                    mgr2.handle
+                        .send_response(
+                            request_id,
+                            Response::HeaderResponse(HeaderResponse::new(vec![])),
+                        )
+                        .await
+                        .unwrap();
+                }
+                _ => panic!("invalid event received {:?}, expected `Request`", event),
+            },
+            _ = tokio::time::sleep(std::time::Duration::from_secs(15)) => {
+                panic!("did not receive `Request in time`");
+            }
```

Comment:
It seems to me `tokio::time::timeout` could be used in place of this select pattern. Would that be an improvement?

Reply:
It would be, thanks!